src/bootstrap/src/ferrocene/doc.rs

```diff
@@ -303,32 +299,35 @@ impl<P: Step> Step for SphinxBook<P> {
                 // To solve them, we first "gather" the object.inv files by building all the documentation
                 // with a special Sphinx builder that doesn't write any other output file, and then we
                 // do a full build of the documents we want.
-                builder.info(&format!("Gathering references of {}", self.src));
+                builder.info(&format!("Gathering references of {}", P::SOURCE));
 
                 // When gathering the objects.inv files, we use a custom Sphinx builder that only
                 // outputs the objects.inv file.
                 cmd.args(&["-b", "ferrocene-intersphinx"]);
+
+                // Avoid circular dependencies.
+                allow_injecting_ids = false;
```

Comment:
When is is set to true? It is initialized as `false`

Reply:
Good catch!